lib/rpmchecksig.cc

```diff
@@ -290,3 +291,61 @@ int rpmcliVerifySignatures(rpmts ts, ARGV_const_t argv)
     rpmKeyringFree(keyring);
     return res;
 }
+
+int cliMatchPubkeys(rpmts ts, ARGV_const_t args, int callback(rpmPubkey, void*), void * userdata)
```

Comment:
Let's keep this baby inside rpmkeys for now.

Reply:
This kinda needs rpmIsValidHex whihc currently is internal only. Sure that's no reason to add this to the API. Still it needs some sort of solution.

Reply:
A low-cost workaround is to drop RPM_GNUC_INTERNAL from rpmIsValidHex(). It'll show up in the library ABI but is basically uncallable because it doesn't have a public header. We have quite a few such items, for similar reasons.

It'll just need an extra directive for rpmkeys in the cmake file to allow it to include from lib/, see rpmlua/rpmgraph for examples on that.

Reply:
I just added my own loop there. Yes, it is sortof duplicated code. But fiddling around with compiler directives doesn't really reduce the complexity either.